simulate/bed.go

```diff
@@ -0,0 +1,57 @@
+package simulate
+
+import (
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/numbers"
+	"sync"
+)
+
+func GoSimulateBed(searchSpace []*bed.Bed, regionCount int, regionLength int64) <-chan bed.Bed {
+	var Length, tmp, chromWindows int64
+	var totalWindows int
+	c := make(chan bed.Bed)
+
+	//count total viable windows
+	for i := 0; i < len(searchSpace); i++ {
+		Length = searchSpace[i].ChromEnd - searchSpace[i].ChromStart
+
+		if Length >= regionLength {
+			totalWindows = totalWindows + int(Length-regionLength)
```

Comment:
Length-regionLength can be negative, but I think we probably want this is evaluate to zero in those cases.  Maybe adding a max function with zero as the other number?  Should Length-regionLength have a "+1"? since there is one place to put a bed of length 5 inside a bed of length 5?

Reply:
The plus one should certainly be there. I didn't think I needed to watch out for a negative Length-regionLength because it's imbedded in an if statement that should control for that